waspc/waspls/src/Wasp/LSP/Commands/ScaffoldTsSymbol.hs

```diff
@@ -0,0 +1,245 @@
+      void $
+        LSP.sendRequest LSP.SWindowShowMessageRequest message $ \case
+          Left err -> logM $ "Error showing message for file created: " <> show err
+          Right (Just (LSP.MessageActionItem "Open")) -> do
+            -- Client selected the "Open" button, ask it to display the file.
+            let showDocument =
+                  LSP.ShowDocumentParams
+                    { _uri = LSP.filePathToUri $ SP.fromAbsFile filepath,
+                      _external = Nothing,
+                      _takeFocus = Just True,
+                      _selection = Nothing
+                    }
+            void $ LSP.sendRequest LSP.SWindowShowDocument showDocument (const (pure ()))
+          Right _ -> return ()
+
+-- | Check if the scaffold command has a template available for the given args.
+-- If this is false, running the command with these args will __definitely__
+-- fail.
+hasTemplateForArgs :: Args -> Bool
+hasTemplateForArgs Args {..} = case P.fileExtension $ SP.toPathAbsFile filepath of
+  Nothing -> False
+  Just ext -> isRight $ templateForRequest pathToExtImport ext
+
+-- | @getTemplateFor pathToExtImport extension@ finds the mustache template in
+-- @data/lsp/templates/ts@ and compiles it.
+getTemplateFor :: MonadIO m => ExprPath -> String -> m (Either String Mustache.Template)
+getTemplateFor exprPath ext = runExceptT $ do
+  templatesDir <- liftIO getTemplatesDir
+  templateFile <- (templatesDir SP.</>) <$> templateForRequest exprPath ext
+  templateExists <- liftIO $ doesFileExist templateFile
+  if templateExists
+    then do
+      compileResult <- liftIO $ Mustache.automaticCompile [SP.fromAbsDir templatesDir] (SP.fromAbsFile templateFile)
+      case compileResult of
+        -- Note: 'error' is used here because all templates should compile succesfully.
+        Left err -> error $ printf "Compilation of template %s failed: %s" (SP.fromAbsFile templateFile) (show err)
+        Right template -> return template
+    else throwError $ printf "No scaffolding template for request: %s does not exist" (SP.fromAbsFile templateFile)
+
+-- | Renders a mustache template to text.
+--
+-- This function is partial: if errors are encountered rendering the template,
+-- @error@ is returned.
+renderTemplate :: Mustache.Template -> Aeson.Value -> Text
+renderTemplate template templateData =
+  let (errs, text) = Mustache.checkedSubstituteValue template $ Mustache.toMustache templateData
+   in if null errs
+        then text
+        else error $ printf "Unexpected errors rendering template: " ++ show errs
+
+data TemplatesDir
+
+data Template
+
+type TemplateFile = SP.Path' (SP.Rel TemplatesDir) (SP.File Template)
+
+templatesDirInDataDir :: SP.Path' (SP.Rel Wasp.Data.DataDir) (SP.Dir TemplatesDir)
+templatesDirInDataDir = [SP.reldir|lsp/templates/ts|]
+
+getTemplatesDir :: IO (SP.Path' SP.Abs (SP.Dir TemplatesDir))
+getTemplatesDir = (SP.</> templatesDirInDataDir) <$> Wasp.Data.getAbsDataDirPath
+
+templateForRequest ::
```

Comment:
This function is actually, byt its arguments, very similar to `getTemplateFor` function above, right?

So it would make sense that they have similar names, and that from their names it is clear how they are similar and also how they are different.